File: resources/messaging/instance_admin.py
import boto3
import logging
import os

logger = logging.getLogger()
try:
    log_level = os.environ["LogLevel"]
    if log_level not in ["INFO", "DEBUG"]:
        log_level = "INFO"
except:
    log_level = "INFO"
logger.setLevel(log_level)

chime = boto3.client("chime-sdk-identity")
ssm = boto3.client("ssm")


def create_app_instance_admin(
    uid,
    appInstanceArn=None,
    appInstanceAdminArn=None,
    **kwargs,
):
    logger.info(f"Creating App Instance Admin: {uid}")
    params = {}
    params["AppInstanceAdminArn"] = appInstanceAdminArn
    params["AppInstanceArn"] = appInstanceArn
    logger.info(f"Params to use: {params}")
    try:
        app_instance_admin = chime.create_app_instance_admin(**params)["AppInstanceAdmin"]
    except Exception as e:
        error = {"error": f"Exception thrown: {e}"}
        logger.error(error)
        raise RuntimeError(error)
    try:
        ssm.put_parameter(
            Name="/chime/appInstanceAdminArn/" + uid,
            Description="appInstanceAdminArn",
            Overwrite=True,
            Value=app_instance_admin["Arn"],
            Type="String",
        )
    except Exception as e:
        error = {"error": f"Exception thrown: {e}"}
        logger.error(error)
        raise RuntimeError(error)

    return app_instance_admin


def delete_app_instance_admin(uid):
    logger.info(f"Deleting an Amazon Chime SDK Messaging Admin: {uid}")

    try:
        ssm.delete_parameter(
            Name="/chime/appInstanceAdminArn/" + str(uid),
        )
    except Exception as e:
        error = {"error": f"Exception thrown: {e}"}
        logger.error(error)
        raise RuntimeError(error)

    return True

File: src/resources/messaging/instanceAdmin.ts
/* eslint-disable import/no-extraneous-dependencies */
import {
  ChimeSDKIdentityClient,
  CreateAppInstanceAdminCommand,
  CreateAppInstanceAdminCommandInput,
  CreateAppInstanceAdminCommandOutput,
} from '@aws-sdk/client-chime-sdk-identity';

import {
  SSMClient,
  DeleteParameterCommand,
  PutParameterCommand,
} from '@aws-sdk/client-ssm';

const chimeClient = new ChimeSDKIdentityClient({
  region: process.env.AWS_REGION,
});

const ssmClient = new SSMClient({ region: process.env.AWS_REGION });

interface AppInstanceAdminProps {
  appInstanceArn?: string;
  appInstanceAdminArn?: string;
}

let createAppInstanceAdminCommandInput: CreateAppInstanceAdminCommandInput;
let createAppInstanceAdminCommandOutput: CreateAppInstanceAdminCommandOutput;

export const CreateAppInstanceAdmin = async (
  uid: string,
  props: AppInstanceAdminProps,
) => {
  createAppInstanceAdminCommandInput = {
    AppInstanceArn: props.appInstanceArn,
    AppInstanceAdminArn: props.appInstanceAdminArn,
  };

  try {
    createAppInstanceAdminCommandOutput = await chimeClient.send(
      new CreateAppInstanceAdminCommand(createAppInstanceAdminCommandInput),
    );
  } catch (error) {
    if (error instanceof Error) {
      console.error(error);
      throw error;
    }
  }

  try {
    await ssmClient.send(
      new PutParameterCommand({
        Name: `/chime/appInstanceAdminArn/${uid}`,
        Description: 'appInstanceAdminArn',
        Value: createAppInstanceAdminCommandOutput.AppInstanceAdmin?.Arn,
        Overwrite: true,
        Type: 'String',
      }),
    );
  } catch (error) {
    if (error instanceof Error) {
      console.error(error);
      throw error;
    }
  }

  return {
    AppInstanceAdminArn:
      createAppInstanceAdminCommandOutput.AppInstanceAdmin?.Arn,
    AppInstanceAdminName:
      createAppInstanceAdminCommandOutput.AppInstanceAdmin?.Name,
  };
};

export const DeleteAppInstanceAdmin = async (uid: string) => {
  try {
    await ssmClient.send(
      new DeleteParameterCommand({ Name: `/chime/appInstanceAdminArn/${uid}` }),
    );
  } catch (error) {
    if (error instanceof Error) {
      console.error(error);
      throw error;
    }
  }
};
